runtime/src/inflation_rewards/mod.rs

```diff
@@ -29,15 +26,14 @@ struct CalculatedStakeRewards {
 // returns a tuple of (stakers_reward,voters_reward)
 pub fn redeem_rewards(
     rewarded_epoch: Epoch,
-    stake_state: StakeStateV2,
-    stake_account: &mut AccountSharedData,
```

Comment:
Just to make sure I'm not missing anything: is the reason why you wrote this fn this way in the first place (https://github.com/solana-labs/solana/pull/20432) so that the account update and StakeStateV2 modifications happened in the same spot? Ergo, now that the pre-PER-feature code has been removed and distribution is done later in all cases, we don't need to modify the `stake_account` at all.

Reply:
The change in https://github.com/solana-labs/solana/pull/20432 was actually similar to this one because it was trying to save an unnecessary deserialization. We already had the deserialized stake state available but we were deserializing it again from the stake account data.

The reason we had stake account passed into this function historically was because we used to update stake accounts in accounts db at the epoch boundary. Since we don't do that anymore, we only need the stake state to do the rewards calculations.

Reply:
Okay, that jives with what I wrote.